container-search/src/main/java/com/yahoo/search/handler/SearchHandler.java

```diff
@@ -552,4 +594,37 @@ public SearchChainRegistry getSearchChainRegistry() {
         return searchChainRegistry;
     }
 
+
+    private void createRequestMapping(Inspector inspector, Map<String, String> map, String parent){
+        inspector.traverse((ObjectTraverser) (key, value) -> {
+            String delimiter = parent.equals("") ? "" : ".";
+            switch (value.type()) {
+                case BOOL:
+                    map.put(parent + delimiter + key, Boolean.toString(value.asBool()));
+                    break;
+                case DOUBLE:
+                    map.put(parent + delimiter + key, Double.toString(value.asDouble()));
+                    break;
+                case LONG:
+                    map.put(parent + delimiter + key, Long.toString(value.asLong()));
+                    break;
+                case STRING:
+                    map.put(parent + delimiter + key, value.asString());
+                    break;
+                case OBJECT:
+                    if (key.equals("grouping")) {
+                        createRequestMapping(value, map, "");
+                    } else {
+                        createRequestMapping(value, map, String.join(delimiter, parent, key));
```

Comment:
Simpler if you define `String qualifiedKey = parent + key` on L600, use that as the map keys through L616, and pass `qualifiedKey + "."` as the third argument on L618?

Reply:
Well, not a big deal :p

Reply:
Oh, yeah. Nice 👍